services/wallet/router/router_v2.go

```diff
@@ -541,38 +542,57 @@ func (r *Router) getBalanceMapForTokenOnChains(ctx context.Context, input *Route
 
 	balanceMap = make(map[string]*big.Int)
 
+	chainError := func(chainId uint64, token string, intErr error) {
+		if err == nil {
+			err = fmt.Errorf("chain %d, token %s: %w", chainId, token, intErr)
+		} else {
+			err = fmt.Errorf("%s; chain %d, token %s: %w", err.Error(), chainId, token, intErr)
+		}
+	}
+
 	for _, chain := range selectedFromChains {
+		// check token existance
 		token := input.SendType.FindToken(r.tokenManager, r.collectiblesService, input.AddrFrom, chain, input.TokenID)
 		if token == nil {
+			chainError(chain.ChainID, token.Symbol, ErrTokenNotFound)
```

Comment:
A point of improvement for a follow-up is to apply the circuit breaker pattern, thus, if the input params are the same, we wouldn't even try to use chains that failed previously (within a small window of time of course).

Reply:
@ilmotta thanks for the proposal. Yes, if we would implement that within a very small time window it would make sense, but, here is another story, actually its two aspects are:
- providers are that flaky (for some reason) that you can in two consecutive calls  once get the data and next  time don't, or vice versa,
- another point is, that this call to prepare the balance map is called at the beginning of the router algorithm and that once for each call for generating the best route, so not sure even if we would implement a circuit breaker if it would pay off since the time between two consecutive calls if for sure more than a second (depends on input params), even more, and many things might be changed within that time window.